R/simulate_infections.R

```diff
@@ -1,19 +1,191 @@
+##'
+##' Simulations are done from given initial infections and, potentially
+##' time-varying, reproduction numbers. Delays and parameters of the observation
+##' model can be specified using the same options as in [estimate_infections()].
+##' @param R a data frame of reproduction numbers (column `R`) by date (column
+##'   `date`). R must be numric and date must be in date format. If any dates
+##'   are missing, it will be assumed that R stays the same
+##' @param initial_infections numeric; the initial number of infections.
+##' @param day_of_week_effect either `NULL` (no day of the week effect) or a
+##'   numerical vector of length specified in [obs_opts()] as `week_length`
+##'   (default: 7) if `week_effect` is set to TRUE. Each element of the vector
+##'   gives the weight given to reporting on this day (normalised to 1).
+##'   The default is `NULL`.
+##' @param estimates deprecated; use [forecast_infections()] instead
+##' @param ... deprecated; only included for backward compatibility
+##' @inheritParams estimate_infections
+##' @inheritParams rt_opts
+##' @importFrom lifecycle deprecate_warn
+##' @importFrom data.table data.table merge.data.table nafill rbindlist
+##' @return A data.table of simulated infections (variable `infections`) and
+##'   reported cases (variable `reported_cases`) by date.
+##' @author Sebastian Funk
+##' @export
+#' @examples
+#' \donttest{
+#'   R <- data.frame(
+#'     date = seq.Date(as.Date("2023-01-01"), length.out = 14, by = "day"),
+#'     R = c(rep(1.2, 7), rep(0.8, 7))
+#'   )
+#'   sim <- simulate_infections(
+#'     R = R,
+#'     initial_infections = 100,
+#'     generation_time = generation_time_opts(
+#'       fix_dist(example_generation_time)
+#'     ),
+#'     delays = delay_opts(fix_dist(example_reporting_delay)),
+#'     obs = obs_opts(family = "negbin")
+#'   )
+#' }
+simulate_infections <- function(estimates, R, initial_infections,
+                                day_of_week_effect = NULL,
+                                generation_time = generation_time_opts(),
+                                delays = delay_opts(),
+                                truncation = trunc_opts(),
+                                obs = obs_opts(),
+                                CrIs = c(0.2, 0.5, 0.9),
+                                pop = 0, ...) {
+  ## deprecated usage
+  if (!missing(estimates)) {
     deprecate_warn(
       "2.0.0",
-      "simulate_infections()",
+      "simulate_infections(estimates)",
       "forecast_infections()",
-      "A new [simulate_infections()] function for simulating from given ",
-      "parameters is planned for implementation in the future."
+      details = paste0(
+        "This `estimates` option will be removed from [simulate_infections()] ",
+        "in version 2.1.0."
+      )
     )
-   forecast_infections(...)
+    return(forecast_infections(estimates = estimates, ...))
+  }
+
+  ## check inputs
+  assert_data_frame(R)
```

Comment:
You can also check `any.missing` (default is TRUE), `min.rows`, `min.cols`, `ncols`, `col.names`.

Reply:
`any.missing` here will check for explicit missingness. I see implicit NA's are filled below.

Reply:
Thanks, improved in 57dd4e18